core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -738,22 +748,34 @@ private void validateGetWithIndexResult(
             .build();
 
     LinkedHashMap<Key, TransactionResult> results = new LinkedHashMap<>(1);
-    TableMetadata tableMetadata = getTableMetadata(scanWithIndex);
-    originalResult.ifPresent(
-        r -> results.put(new Snapshot.Key(scanWithIndex, r, tableMetadata), r));
+    originalResult.ifPresent(r -> results.put(new Snapshot.Key(scanWithIndex, r, metadata), r));
 
     // Validate the result to check if there is no anti-dependency
     validateScanResults(storage, scanWithIndex, results, false);
   }
 
   private void validateGetResult(
-      DistributedStorage storage, Get get, Optional<TransactionResult> originalResult)
+      DistributedStorage storage,
+      Get get,
+      Optional<TransactionResult> originalResult,
+      TableMetadata metadata)
       throws ExecutionException, ValidationConflictException {
-    // Only get the tx_id column because we use only them to compare
-    get = Get.newBuilder(get).clearProjections().projection(Attribute.ID).build();
-
     // Check if a read record is not changed
-    Optional<TransactionResult> latestResult = storage.get(get).map(TransactionResult::new);
+    Optional<TransactionResult> latestResult =
+        storage
+            .get(ConsensusCommitUtils.prepareGetForStorage(get, metadata))
```

Comment:
Use `ConsensusCommitUtils.prepareGetForStorage()` to use the same Get as that used for transaction reads.